src/services/hub-service.ts

```diff
@@ -0,0 +1,124 @@
+    } catch {
+      throw new CLIError('Hub is not accessible')
+    }
+
+    CliUx.ux.action.stop()
+
+    const selectedBundleGroup = await this.promptSelectBundleGroup()
+    console.log(`You have selected bundle group "${selectedBundleGroup?.bundleGroupName}".`)
+
+    CliUx.ux.action.start(`Opening bundle group "${selectedBundleGroup.bundleGroupName}"`)
+    const bundles: Bundle[] = await this.hubApi.getBundlesByBundleGroupId(selectedBundleGroup.bundleGroupVersionId)
+    CliUx.ux.action.stop()
+
+    const selectedBundle: Bundle = await this.promptBundleSelect(bundles, selectedBundleGroup)
+
+    return { selectedBundleGroup, selectedBundle }
+  }
+
+  private async initCloneBundle(selectedBundle: Bundle): Promise<string> {
+    const newBundleName = await CliUx.ux.prompt('New bundle name')
+
+    this.checkBundleName(newBundleName)
+    this.checkBundleDirectory(newBundleName)
+
+    CliUx.ux.action.start(`Downloading bundle ${selectedBundle.bundleName}`)
+    try {
+      // Using stdio 'pipe' option to print stderr only through CLIError
+      cp.execSync(`git clone ${selectedBundle.gitSrcRepoAddress} ./${newBundleName}`, { stdio: 'pipe' })
+    } catch (error) {
+      throw new CLIError(error as Error)
+    }
+
+    CliUx.ux.action.stop()
+
+    const newBundleVersion = await CliUx.ux.prompt('What\'s the version number of this bundle?')
+
+    CliUx.ux.action.start('Making changes to the bundle descriptor')
+    this.renameBundleDescriptor(newBundleName, newBundleVersion)
+    CliUx.ux.action.stop()
+
+    this.removeGitInfo(newBundleName)
+    this.initGitRepo(newBundleName)
+
+    return newBundleName
+  }
+
+  private async checkMissingDirectories(bundleName: string) {
+    this.createSubDirectoryIfNotExist(bundleName, 'microservices')
+    this.createSubDirectoryIfNotExist(bundleName, 'microfrontends')
+    this.createSubDirectoryIfNotExist(bundleName, '.ent')
+    this.createSubDirectoryIfNotExist(bundleName, '.ent', 'output')
+    this.createSubDirectoryIfNotExist(bundleName, 'epc')
+  }
+
+  private async promptSelectBundleGroup(): Promise<BundleGroup> {
+    const choices = this.loadedBundleGroups.map(bundleGroup => ({ name: bundleGroup.bundleGroupName, value: bundleGroup }))
+    const response: any = await inquirer.prompt([{
+      name: 'bundlegroup',
+      message: 'Select a bundle group',
+      type: 'list',
+      choices,
+    }])
+    return response.bundlegroup;
+  }
+
+  private async promptBundleSelect(bundleChoices: Bundle[], selectedBundleGroup: BundleGroup): Promise<Bundle> {
+    const choices = bundleChoices.map(bundle => ({ name: bundle.bundleName, value: bundle  }))
+    const response: any = await inquirer.prompt([{
+      name: 'bundle',
+      message: `Select a bundle from ${selectedBundleGroup.bundleGroupName}`,
+      type: 'list',
+      choices,
+    }])
+    return response.bundle;
+  }
+
+  private renameBundleDescriptor(name: string, version: string): void {
```

Comment:
This is responsiblity of `BundleDescriptorService`

Reply:
this has been moved to initializer service and refactored as well